### PR TITLE
fix: added types FiltersHelper Service

### PR DESCRIPTION
**Pull request by CODEX108**
### <samp>🤖 Generated by Copilot at cc70e48</samp>

This pull request enhances the type safety and readability of the code related to the filters helper service. It uses the `AdvancesStates` enum, adds explicit types to variables and parameters, and fixes implicit any errors in `filters-helper.service.ts` and `filters-helper.service.spec.ts`.

### <samp>🤖 Generated by Copilot at cc70e48</samp>

> _To make filters helper more robust_
> _We imported the `AdvancedStates` enum_
> _And added explicit types_
> _To avoid any plights_
> _With implicit any or a bad outcome_

### <samp>🤖 Generated by Copilot at cc70e48</samp>

*  Change the type of the `testSelectedFilter` and `testSelectedFilters` variables from `any` to `string | string[] | number` in the test cases of the `FiltersHelperService` to make them more specific and avoid implicit any errors. ([link](https://github.com/fylein/fyle-mobile-app/pull/2111/files?diff=unified&w=0#diff-652ea9cee829c082b074a1b9963b62be72217acd88e41578a0c34af5827d4affL266-R266), [link](https://github.com/fylein/fyle-mobile-app/pull/2111/files?diff=unified&w=0#diff-652ea9cee829c082b074a1b9963b62be72217acd88e41578a0c34af5827d4affL281-R281), [link](https://github.com/fylein/fyle-mobile-app/pull/2111/files?diff=unified&w=0#diff-652ea9cee829c082b074a1b9963b62be72217acd88e41578a0c34af5827d4affL306-R306), [link](https://github.com/fylein/fyle-mobile-app/pull/2111/files?diff=unified&w=0#diff-652ea9cee829c082b074a1b9963b62be72217acd88e41578a0c34af5827d4affL331-R331), [link](https://github.com/fylein/fyle-mobile-app/pull/2111/files?diff=unified&w=0#diff-652ea9cee829c082b074a1b9963b62be72217acd88e41578a0c34af5827d4affL347-R347), [link](https://github.com/fylein/fyle-mobile-app/pull/2111/files?diff=unified&w=0#diff-652ea9cee829c082b074a1b9963b62be72217acd88e41578a0c34af5827d4affL363-R363))
* Import the `AdvancesStates` enum from the `advances-states.model.ts` file and use it as a type for the filters related to the state of the advances in the `FiltersHelperService`. ([link](https://github.com/fylein/fyle-mobile-app/pull/2111/files?diff=unified&w=0#diff-3e15dc05566eb2c216aca7c76d0c024a6301c96bf638401467cb0ffb22986459R12), [link](https://github.com/fylein/fyle-mobile-app/pull/2111/files?diff=unified&w=0#diff-3e15dc05566eb2c216aca7c76d0c024a6301c96bf638401467cb0ffb22986459L66-R79), [link](https://github.com/fylein/fyle-mobile-app/pull/2111/files?diff=unified&w=0#diff-3e15dc05566eb2c216aca7c76d0c024a6301c96bf638401467cb0ffb22986459L94-R95))
* Give the `filterPillsMap` and `filtersMap` objects a type of `Record<string, string>` to make them more explicit and avoid implicit any errors in the `FiltersHelperService`. ([link](https://github.com/fylein/fyle-mobile-app/pull/2111/files?diff=unified&w=0#diff-3e15dc05566eb2c216aca7c76d0c024a6301c96bf638401467cb0ffb22986459L22-R23), [link](https://github.com/fylein/fyle-mobile-app/pull/2111/files?diff=unified&w=0#diff-3e15dc05566eb2c216aca7c76d0c024a6301c96bf638401467cb0ffb22986459L66-R79))
* Change the `convertDataToFilters` method to accept an array of `SelectedFilters<string | string[] | number>` instead of `any` and cast the `stateFilter.value` and `sortBy.value` to `AdvancesStates[]` and `AdvancesStates` respectively in the `FiltersHelperService`. ([link](https://github.com/fylein/fyle-mobile-app/pull/2111/files?diff=unified&w=0#diff-3e15dc05566eb2c216aca7c76d0c024a6301c96bf638401467cb0ffb22986459L59-R60), [link](https://github.com/fylein/fyle-mobile-app/pull/2111/files?diff=unified&w=0#diff-3e15dc05566eb2c216aca7c76d0c024a6301c96bf638401467cb0ffb22986459L66-R79))
* Cast the `data` returned from the `filterPopover.onWillDismiss()` method to an array of `SelectedFilters<string | number | string[]>` to match the expected type of the `convertDataToFilters` method in the `FiltersHelperService`. ([link](https://github.com/fylein/fyle-mobile-app/pull/2111/files?diff=unified&w=0#diff-3e15dc05566eb2c216aca7c76d0c024a6301c96bf638401467cb0ffb22986459L115-R116))

## Clickup
https://app.clickup.com/t/85zt699fp

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/core/services/filters-helper.service.ts [hunk 1]
```diff
@@ -56,26 +57,26 @@ export class FiltersHelperService {
...
-  convertDataToFilters(selectedFilters: SelectedFilters<any>[]): Filters {
+  convertDataToFilters(selectedFilters: SelectedFilters<string | string[] | number>[]): Filters {
     const generatedFilters: Filters = {};
 
     const stateFilter = selectedFilters.find((filter) => filter.name === 'State');
     const sortBy = selectedFilters.find((filter) => filter.name === 'Sort By');
 
     if (stateFilter) {
-      generatedFilters.state = stateFilter.value;
+      generatedFilters.state = stateFilter.value as AdvancesStates[];
```
Comment (Dimple16) [inline src/app/core/services/filters-helper.service.ts:67]: why use `as AdvancesStates[];` here? check this - https://fylein.slack.com/archives/C049CK3JPJQ/p1686117358172619

File: src/app/core/services/filters-helper.service.ts [hunk 2]
```diff
@@ -56,26 +57,26 @@ export class FiltersHelperService {
...
     }
 
     if (sortBy && sortBy.value) {
-      generatedFilters.sortParam = this.getSortParam(sortBy.value);
-      generatedFilters.sortDir = this.getSortDir(sortBy.value);
+      generatedFilters.sortParam = this.getSortParam(sortBy.value as AdvancesStates);
```
Comment (Dimple16) [inline src/app/core/services/filters-helper.service.ts:71]: same as above

**Review by Dimple16 — CHANGES_REQUESTED** [review]
added comments

File: src/app/core/services/filters-helper.service.ts [hunk 3]
```diff
@@ -56,11 +57,15 @@ export class FiltersHelperService {
     return filterPills;
   }
 
-  convertDataToFilters(selectedFilters: SelectedFilters<any>[]): Filters {
+  convertDataToFilters(selectedFilters: SelectedFilters<string | AdvancesStates[]>[]): Filters {
```
Comment (yksolanki9) [inline src/app/core/services/filters-helper.service.ts:60]: Didnt find any usage of number type here so removed it

**Pull request closed without merging**